web/common.inc.sh

```diff
@@ -32,6 +31,34 @@ compile ( ) {
   fi
 }
 
+_copy_dir_if_exists ( ) {
```

Comment:
This could be a problem for CI, where we would want it to fail if something was missing?

Reply:
If something's missing, it'd be because either one of the TS projects failed to build or because a resource copy to `build/app/resources` failed.  Something else would fail before this would have the chance to fail in any situation where this "should" fail.

Or because I got this function wrong... that _would_ be a possibility at the moment.

Reply:
I guess the principle of defence in depth applies.